src/burrow/burrow.js

```diff
@@ -0,0 +1,157 @@
+     * @async
+     */
+    getContext(name, args) {
+        let config  = require(this.configPath);
+        let context = config.burrow.context;
+        if(typeof context === 'undefined') {
+
+            let grpc = config.burrow.network.validator.grpc;
+            if(grpc === null) {
+                logger.error('Error: Validator url not set.');
+            }
+            let account = fs.readFileSync(util.resolvePath(config.burrow.network.validator.address)).toString();
+
+            logger.info(`Account: ${account}`);
+            logger.info(`GRPC: ${grpc}`);
+
+            let options = {objectReturn: true};
+            let burrow = monax.createInstance(grpc, account, options);
+            context = {stream: burrow, account: account};
+        }
+        return Promise.resolve(context);
+    }
+
+    /**
+     * Release the given Burrow context.
+     * @param {object} context The Burrow context to release.
+     * @async
+     */
+    async releaseContext(context) {
+        // nothing to do
+    }
+
+    /**
+   * Invoke a smart contract.
+   * @param {Object} context context object
+   * @param {String} contractID identity of the contract
+   * @param {String} contractVer version of the contract
+   * @param {Array} args array of JSON formatted arguments for multiple transactions
+   * @param {Number} timeout request timeout, in seconds
+   * @return {Promise<object>} the promise for the result of the execution.
+   */
+    async invokeSmartContract(context, contractID, contractVer, args, timeout) {
+        let promises = [];
+        args.forEach((item, index)=>{
+            promises.push(this.burrowTransaction(context, contractID, contractVer, item, timeout));
+        });
+        return await Promise.all(promises);
+    }
+
+    /**
+   * Submit a transaction to the burrow daemon with the specified options.
+   * @param {Object} context context object
+   * @param {String} contractID identity of the contract
+   * @param {String} contractVer version of the contract
+   * @param {Array} args array of JSON formatted arguments for multiple transactions
+   * @param {Number} timeout request timeout, in seconds
+   * @return {Promise<TxStatus>} result and stats of the transaction invocation.
+   */
+    async burrowTransaction(context, contractID, contractVer, args, timeout) {
+        let status = new TxStatus(args.account);
+        status.Set('timeout', timeout*1000);
+        if(context.engine) {
+            context.engine.submitCallback(1);
+        }
+
+        let tx = {
+            Input: {
+                Address: Buffer.from(context.account,'hex'),
+                Amount: args.money
+            },
+            GasLimit: 5000,
+            Fee: 5000
+        };
+
+        return context.stream.transact.CallTxSync(tx).then((execution)=>{
+            status.SetID(execution.TxHash.toString());
+            status.SetStatusSuccess();
+            return status;
+        });
```

Comment:
You should add a `.catch(err => {...})` to the chain in case an error occurs, so you can set the status to failed and "hide" the error from the engine. Or the following would be even better:

```js
try {
  let execution = await context.stream.transact.CallTxSync(tx);
  status.SetID(execution.TxHash.toString());
  status.SetStatusSuccess();
} catch (err) {
  // log error
  status.SetStatusFail();
}

return status;
```